### PR TITLE
Glt 269 dao test

### PR DESCRIPTION
**Pull request by djcooke**
* Set up for DAO testing
* Tests written for SQLRunDAO for example usage
* Some bugs fixed in SQLRunDAO

### CODE REVIEW

**Comment by apmasell** [comment]
Looks Groovy to me. I'm not certain we need Groovy to do that; it looks like porting it to Java would be reasonable. I have no strong opinion though.

**Comment by djcooke** [comment]
How do you propose packaging/running it if it were ported to Java? Using a Groovy script with groovy-maven-plugin seemed a lot simpler than alternatives I looked at

**Comment by apmasell** [comment]
No idea; if that's the reason, then go for it. I'm assuming this `exec-maven-plugin` is complicated. Sigh. I just expect build system to know how to run things.

File: sqlstore/pom.xml [hunk 1]
```diff
@@ -176,6 +195,22 @@
...
     <dependency>
+      <groupId>com.h2database</groupId>
+      <artifactId>h2</artifactId>
+      <version>1.4.190</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-all</artifactId>
+      <version>1.10.19</version>
```
Comment (frankie13) [inline sqlstore/pom.xml]: perhaps a `<scope>test</scope>` here?

Comment (djcooke) [inline sqlstore/pom.xml]: Definitely, will add

**Comment by frankie13** [comment]
I get the following message when I try to `mvn clean install -P external`.  `mvn clean install` seems to work fine though...
:space_invader: :space_invader: :space_invader: :space_invader: :space_invader: :space_invader: 
    [ERROR] BUILD ERROR
    [INFO] ------------------------------------------------------------------------
    [INFO] Internal error in the plugin manager executing goal 'org.codehaus.gmaven:groovy-maven-plugin:2.0:execute': Unable to find the mojo 'execute' (or one of its required components) in the plugin 'org.codehaus.gmaven:groovy-maven-plugin'
    Component descriptor cannot be found in the component repository: org.codehaus.plexus.PlexusContainer.

File: sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLRunDAOTest.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+  }
+  
+  private void assertNonLazyThings(Run run) {
+    assertNotNull(run);
+    assertFalse(run.getSequencerPartitionContainers().isEmpty());
+    assertFalse(run.getRunQCs().isEmpty());
+    assertFalse(run.getNotes().isEmpty());
+  }
+
+}
```
Comment (frankie13) [inline sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLRunDAOTest.java]: perfection! :white_check_mark:

**Comment by djcooke** [comment]
Fixed build. I used relative paths in the groovy script, so it was building fine from the sqlstore directory, but not from the parent directory

File: sqlstore/pom.xml (hunk 1 above)
Comment (djcooke) [inline sqlstore/pom.xml]: Added

**Comment by frankie13** [comment]
:whale: I still get an error when doing a `mvn clean install -P external`

Internal error in the plugin manager executing goal 'org.codehaus.gmaven:groovy-maven-plugin:2.0:execute': Unable to find the mojo 'execute' (or one of its required components) in the plugin 'org.codehaus.gmaven:groovy-maven-plugin'
Component descriptor cannot be found in the component repository: org.codehaus.plexus.PlexusContainer.

**Comment by djcooke** [comment]
I take it you're using Maven 2? I've been using Maven 3 without issue, but I just tried building with Maven 2 and I get this error as well. The plugin is supposed to be compatible with Maven 2.0+. I'll investigate further

**Pull request merged**